quam_builder/architecture/quantum_dots/README.md

```diff
@@ -0,0 +1,682 @@
+
+- NOTE: When using QUA loops (such as for_, or infinite_loop_) bear in mind that the internal voltage tracker can not track voltages past the first loop. As such, ensure that the start and end point of the loop of the elements are identical
+
+**Creating a `VoltageSequence`:**
+
+- The sequence must be defined within a QUA program. 
+
+
+**Core Methods (used in `qua.program()` context):**
+
+- `step_to_voltages(voltages: Dict[str, float], duration: int)`  
+  Steps all specified channels directly to the given voltage levels and holds them for the specified duration (in nanoseconds). This creates immediate voltage changes without ramping. Both `voltages` values and `duration` can be QUA variables for dynamic control.
+
+  ```python
+  voltage_seq.step_to_voltages(voltages={"P1": 0.3, "P2": 0.1}, duration=1000)
+  ```
+
+- `ramp_to_voltages(voltages: Dict[str, float], duration: int, ramp_duration: int)`  
+  Ramps all specified channels to the given voltage levels over the specified ramp duration, then holds them for the duration (both in nanoseconds). This provides smooth voltage transitions useful for avoiding voltage spikes that could affect sensitive quantum systems. All parameters can be QUA variables.
+
+  ```python
+  voltage_seq.ramp_to_voltages(voltages={"P1": 0.0}, duration=500, ramp_duration=40)
+  ```
+
+- `step_to_point(name: str, duration: Optional[int] = None)`  
+  Steps all channels to the voltages defined in a predefined `VoltageTuningPoint` macro. If no duration is provided, uses the default duration from the tuning point definition. This enables quick transitions to well-defined system states. The `duration` parameter can be a QUA variable.
+
+  ```python
+  voltage_seq.step_to_point("idle")
+  voltage_seq.step_to_point("readout", duration=2000)  # Override default duration
+  ```
+
+- `ramp_to_point(name: str, ramp_duration: int, duration: Optional[int] = None)`  
+  Ramps all channels to the voltages defined in a predefined `VoltageTuningPoint` over the specified ramp duration, then holds them. Combines the smooth transitions of ramping with the convenience of predefined voltage states. Both `ramp_duration` and `duration` can be QUA variables.
+
+  ```python
+  voltage_seq.ramp_to_point("idle", ramp_duration=50, duration=1000)
+  ```
+
+- `ramp_to_zero(ramp_duration: Optional[int] = None)`  
+  Ramps the voltage on all channels in the GateSet to zero and resets the integrated voltage tracking for each channel. If no duration is specified, uses QUA's built-in `ramp_to_zero` command for immediate ramping. Essential for safely returning to a neutral state. The `ramp_duration` parameter can be a QUA variable.
+
+  ```python
+  voltage_seq.ramp_to_zero()  # Immediate ramp using QUA built-in
+  voltage_seq.ramp_to_zero(ramp_duration=100)  # Controlled ramp over 100ns
+  ```
```

Comment:
All of these are indeed the intended behavior, do you think that we'll manage to validate everything before merging, or it should be part of a different PR?